model/models/models.py

```diff
@@ -0,0 +1,82 @@
+from pydantic import BaseModel, ConfigDict
+from abc import ABC, abstractmethod
+import pandas as pd
+from sklearn.linear_model import LinearRegression
+from xgboost import XGBRegressor
+from keras.api.models import Sequential
+from keras.api import Layer
+from sklearn.metrics import mean_squared_error, r2_score
+from typing import Union
+
+
+class CustomBaseModel(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+
+class Model(ABC, CustomBaseModel):
+    model: Union[LinearRegression, XGBRegressor, Sequential]
+
+    @abstractmethod
+    def fit(self, X: pd.DataFrame, y: pd.DataFrame):
+        pass
+
+    @abstractmethod
+    def predict(self, X: pd.DataFrame):
+        pass
+
+    @abstractmethod
+    def evaluate(self, predictions: pd.DataFrame, y: pd.DataFrame):
+        pass
+
+
+class Regressor(Model):
+    def __init__(self, **kwargs):
+        super().__init__(model=LinearRegression())
+
+    def fit(self, X: pd.DataFrame, y: pd.DataFrame) -> None:
+        self.model.fit(X, y)
+
+    def predict(self, X: pd.DataFrame) -> pd.DataFrame:
+        return self.model.predict(X)
+
+    def evaluate(self, X, y: pd.DataFrame):
```

Comment:
Missing return type hint.
```suggestion
    def evaluate(self, X, y: pd.DataFrame) -> float:
```